File: samples/C# Samples/ProcessScannedDocumentsWithBarcodeSeparators/ProcessScannedDocumentsWithBarcodeSeparators/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PSDWBS
{
    public partial class Form2 : Form
    {

        private string m_PDFFileName = null;
        private string m_FolderPath = null;
        public Form2()
        {
            InitializeComponent();
        }
        public Form2(string strFolderPath, string strPDFName)
        {
            InitializeComponent();
            this.Text = "Set PDF File Name";
            m_PDFFileName =strPDFName;
            tbxPDFName.Text = m_PDFFileName+".pdf";
            m_FolderPath = strFolderPath;
        }
        private void tbxSetFileName_Click(object sender, EventArgs e)
        {
            if (tbxPDFName.Text != null)
            {
                string strNameText = null;
                strNameText = tbxPDFName.Text;
                foreach (char text in strNameText )
                {
                    foreach (char temp in System.IO.Path.GetInvalidFileNameChars())
                    {
                        if (text == temp)
                        {
                            MessageBox.Show("The name of  PDF file contains  illegal character!", "Error");
                            return;
                        }

                    }
                }
                string FilePath = m_FolderPath + "\\" + strNameText ;
                if (System.IO.File.Exists(FilePath))
                {
                    MessageBox.Show("The name of PDF file has exists!","Warning");
                    return;
                }
                m_PDFFileName = tbxPDFName.Text;
            }
            else
            {
                MessageBox.Show(" The name of PDF file can not be null!","Warning");
                return;
            }
            this.Close();
        }
        public string GetPDfFileName()
        {
            return m_PDFFileName;
        }
    }
}

File: samples/VB .NET Samples/SeparateDocumentByBarcode/SeparateDocumentByBarcode/Form2.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class Form2
	Inherits Form
	Private m_PDFFileName As String = Nothing
	Private m_FolderPath As String = Nothing
	Public Sub New()
		InitializeComponent()
	End Sub

	Public Sub New(strFolderPath As String, strPDFName As String)
		InitializeComponent()
		Me.Text = "Set PDF File Name"
		m_PDFFileName = strPDFName
		tbxPDFName.Text = m_PDFFileName & ".pdf"
		m_FolderPath = strFolderPath
	End Sub
	Private Sub tbxSetFileName_Click(sender As Object, e As EventArgs)
		If tbxPDFName.Text IsNot Nothing Then
			Dim strNameText As String = Nothing
			strNameText = tbxPDFName.Text
			For Each text As Char In strNameText
				For Each temp As Char In System.IO.Path.GetInvalidFileNameChars()
					If text = temp Then
						MessageBox.Show("The name of  PDF file contains  illegal character!", "Error")
						Return

					End If
				Next
			Next
			Dim FilePath As String = m_FolderPath & "\" & strNameText
			If System.IO.File.Exists(FilePath) Then
				MessageBox.Show("The name of PDF file has exists!", "Warning")
				Return
			End If
			m_PDFFileName = tbxPDFName.Text
		Else
			MessageBox.Show(" The name of PDF file can not be null!", "Warning")
			Return
		End If
		Me.Close()
	End Sub
	Public Function GetPDfFileName() As String
		Return m_PDFFileName
	End Function
End Class
